vassal-app/src/main/java/VASSAL/build/module/GameRefresher.java

```diff
@@ -356,7 +356,7 @@ private void processGamePiece(GamePiece piece, Command command) {
     final Stack oldStack = piece.getParent();
     if (oldStack == null) {
       final Immobilized i = (Immobilized)Decorator.getDecorator(piece, Immobilized.class);
-      if (i == null) {
+      if (i == null || !i.isCanStack()) {
```

Comment:
Shouldn't the whole search for Immobilized be removed, and instead just check `Boolean.TRUE.equals(piece.getProperty(NO_STACK))`?

Everywhere else we check for stacking we do so via the property. Which will help when I introduce Mat returning a true to "NO_STACK".